Adding Scenario Documentation Policy and RACI Chart, additional documenation cleanup

calebevans opened the pull request:
**New Files:**

- `RACI_Chart.md` : A Markdown table of the purposed CSPI OpenShift CI RACI chart
- `Markdown_Resources.md` : A list of resources for using Markdown that is referenced in the documentation policies
- `Scenario_Documentation_Policy.md` : A document outlining the purposed CSPI scenario documentation policy

**Modified Files**

- `Container_Creation_Guide.md` : Modified to refer to the "Interop Team" as the "CSPI Team". Also removed the use of "Integrated Product" to use "Layered Product" instead.
- `Step_Registry_Documentation_Policy.md` : Modified to refer to the "Interop Team" as the "CSPI Team". Also removed the "Markdown Resources" section and moved that information to its own document so it can be referenced from other documents.
- `SUMMARY.md` : Added a link to the `Markdown_Resources.md` document.

calebevans committed:
Add RACI chart

calebevans committed:
Adding the Scenario Documentation Policy document, separated the markdown resources information to a new file, updated SUMMARY,  replaced Interop with CSPI where it made sense, and replaced integrated product with layered product

calebevans commented on docs/Policy/Documentation/Scenario_Documentation_Policy.md:
```diff
@@ -0,0 +1,114 @@
+- [Conclusion](#conclusion)
+
+## Introduction
+Documenting our scenarios is an important part of the CSPI onboarding and maintenance process. Ensuring we have a comprehensive document for each scenario will help future current and future team members debug and maintain each scenario. Please follow this documentation policy to write helpful documentation for each new scenario.
```
```suggestion
Documenting our scenarios is an important part of the CSPI onboarding and maintenance process. Ensuring we have a comprehensive document for each scenario will help current and future team members debug and maintain each scenario. Please follow this documentation policy to write helpful documentation for each new scenario.
```

calebevans committed:
Update docs/Policy/Documentation/Scenario_Documentation_Policy.md

Fixing a typo

calebevans commented on docs/Policy/Documentation/Scenario_Documentation_Policy.md:
```diff
@@ -0,0 +1,114 @@
+Each scenario written by CSPI within OpenShift CI should have a `README.md` file associated with it in the same folder as the scenario's configuration file. For example the MTR scenario's documentation can be found at `ci-operator/config/windup/windup_integration_test/README.md` along with the configuration YAML file. Documentation should be written in Markdown. If you are unfamiliar with Markdow ...
+
+### Getting Started
+1. Create the `README.md` file in the same director that holds the scenario's configuration YAML file.
```
```suggestion
1. Create the `README.md` file in the same directory that holds the scenario's configuration YAML file.
```

calebevans committed:
Update docs/Policy/Documentation/Scenario_Documentation_Policy.md to fix typo

ascerra commented:
Hm by replacing Interop with CSPI we are losing some context. Everyone could get a general sense about what the Interop team might be doing but when we say the CSPI team this could mean just about anything. I think we'll need to link to a team doc.

This might be the best team doc to use https://source.redhat.com/groups/public/cspi_qe

ascerra commented on docs/Onboarding/RACI_Chart.md:
```diff
@@ -0,0 +1,22 @@
+| Prerequisites       | C       | R          | I                  |
+| Kick-off            | R       | A          | I                  |
+| Triggering          | R       | I          |                    |
+| Scenario Foundation | R       | A          |                    |
```
Change `Scenario Foundation` to `Provision`

calebevans replied:
Fixed

ascerra commented on docs/Onboarding/RACI_Chart.md:
```diff
@@ -0,0 +1,22 @@
+| Document            | R       | A          |                    |
+| Production          | R       | A          | I                  |
+| Maintenance         | C       | R          | I                  |
+| Triggering          | R       | I          |                    |
```
`Triggering` is repeated we can remove this one.

calebevans replied:
Fixed

ascerra commented on docs/Onboarding/RACI_Chart.md:
```diff
@@ -0,0 +1,22 @@
+| Scenario Foundation | R       | A          |                    |
+| Orchestrate         | R       | A          |                    |
+| Execute             | R       | A          |                    |
+| Report              | R       | A          |                    |
```
We should add `Project Management` as C-Consulted for Report.

calebevans replied:
Fixed

ascerra commented on docs/Policy/Documentation/Markdown_Resources.md:
```diff
@@ -0,0 +1,25 @@
+- [Mermaid Documentation](https://mermaid.js.org/intro/)
+  - Used to dynamically generate flow (and other) charts within Markdown
+
+## Editors
```
I know PyCharm is pretty popular which makes me think we can add this here.

calebevans replied:
Added

ascerra commented on docs/Policy/Documentation/Markdown_Resources.md:
```diff
@@ -0,0 +1,25 @@
+- [Mermaid Documentation](https://mermaid.js.org/intro/)
+  - Used to dynamically generate flow (and other) charts within Markdown
+
+## Editors
```
Editors and Plugins seems like it could belong in [Developers_Guide](https://github.com/CSPI-QE/ocp-ci-docs/blob/main/docs/Onboarding/Developers_Guide.md)

Actually this whole file could just be a subsection of the developers guide what to you think about that?

calebevans replied:
Decided in call to keep this separate.

ascerra commented on docs/Policy/Documentation/Scenario_Documentation_Policy.md:
```diff
@@ -0,0 +1,114 @@
+   - **Note:** This plugin will create the Table of Contents for you and keep it automatically updated with hyperlinks upon every save of the document. 
+
+### Adding Documentation
+The outline of these documents should generally follow the following structure:
```
```suggestion
The outline of these documents should generally follow the structure:
```

calebevans replied:
Fixed

ascerra commented on docs/Policy/Documentation/Scenario_Documentation_Policy.md:
````diff
@@ -0,0 +1,114 @@
+<sub><sup>`ci-operator/config/windup/windup_integration_test/README.md`</sup></sub>
+```markdown
+## Purpose
+To provision the necessary infrastructure and using that infrastructure to execute MTR interop tests. The results of theses tests should be reported to the appropriate sources following execution.
````
```suggestion
To provision the necessary infrastructure and use that infrastructure to execute MTR interop tests on pre release OCP builds. The results of theses tests will be reported to the appropriate sources following execution.
```

calebevans replied:
Fixed

ascerra commented on docs/Policy/Documentation/Step_Registry_Documentation_Policy.md:
```diff
@@ -9,16 +9,12 @@
-  - [Plugins](#plugins)
 
 ## Introduction
 Documenting new steps in the step registry is important because it can help use work more efficiently as well as helping users avoid duplicating work in the Step Registry. Duplicated work and lack of documentation, at the time of writing this document, seems to be an issue in the Step Registry and we would like to start solving that issue. When adding a new workflow, ref, chain, _etc._ to the Ste ...
```
```suggestion
Documenting new steps in the step registry is important because it can help us work more efficiently as well as helping users avoid duplicating work in the Step Registry. Duplicated work and lack of documentation, at the time of writing this document, seems to be an issue in the Step Registry and we would like to start solving that issue. When adding a new workflow, ref, chain, _etc._ to the Step Registry, please follow the documentation guide below to write helpful documentation for the new step.
```

ascerra commented on docs/Policy/Documentation/Step_Registry_Documentation_Policy.md:
```diff
@@ -133,23 +129,3 @@ In the `usage` section of this documentation, please try to be specific and give
 OpenShift CI has implemented a great way to autogenerate documentation for a step, but it does not give as a good way to be as specific as we'd like to be in our documentation. The documentation can be found at the following link: https://steps.ci.openshift.org/
```
```suggestion
OpenShift CI has implemented a great way to autogenerate documentation for a step, but its not as specific as we'd like to be in our documentation. The documentation can be found at the following link: https://steps.ci.openshift.org/
```

ascerra commented on docs/Policy/Documentation/Step_Registry_Documentation_Policy.md:
```diff
@@ -133,23 +129,3 @@ In the `usage` section of this documentation, please try to be specific and give
 OpenShift CI has implemented a great way to autogenerate documentation for a step, but it does not give as a good way to be as specific as we'd like to be in our documentation. The documentation can be found at the following link: https://steps.ci.openshift.org/
 
 This autogenerated documentation is build using the `description` items in step configuration files. While helpful, it is hard to be verbose in how you document your step. Simpler steps may be able to be documented in this way, but we find the `README.md` documents to be much more helpful.
```
```suggestion
This autogenerated documentation is built using the `description` items in step configuration files. While helpful, it is hard to be verbose in how you document your step. Simpler steps may be able to be documented in this way, but we find the `README.md` documents to be much more helpful.
```

calebevans committed:
Typo - Change "use" to "us"

Co-authored-by: Adam Scerra <41153661+ascerra@users.noreply.github.com>

calebevans committed:
Wording - Update working around auto-generated documentation

Co-authored-by: Adam Scerra <41153661+ascerra@users.noreply.github.com>

calebevans committed:
Typo - change "build" to "built"

Co-authored-by: Adam Scerra <41153661+ascerra@users.noreply.github.com>

calebevans committed:
Adding "Who is CSPI" on the welcome page

calebevans committed:
Adding adams suggestions

calebevans committed:
Fix typos and add additional context



Fixing a typo
Update docs/Policy/Documentation/Scenario_Documentation_Policy.md to fix typo
Typo - Change "use" to "us"

Co-authored-by: Adam Scerra <41153661+ascerra@users.noreply.github.com>
Wording - Update working around auto-generated documentation

Co-authored-by: Adam Scerra <41153661+ascerra@users.noreply.github.com>
Typo - change "build" to "built"

Co-authored-by: Adam Scerra <41153661+ascerra@users.noreply.github.com>
Adding "Who is CSPI" on the welcome page

Adding adams suggestions

calebevans committed:
Merge remote-tracking branch 'origin/RACI-chart-and-policy' into RACI-chart-and-policy

ascerra commented:
LGTM
